### PR TITLE
Update API/Permissions tests to follow new org/loc taxonomies logic

### PR DESCRIPTION
**Pull request by svtkachenko**
Now 
>  Only administrator can choose empty taxonomies or taxonomies that they aren't assigned to,
>  other users can select only taxonomies they are granted to assign
>  and they can't leave the selection empty (which would mean global resource)

so it is needed to specify same org and loc for non-admin user and created entity. Also I've added one katello entity to increase coverage.

Thanks to @ldjebran for the idea how to handle both types of entities (with loc and org and without them). :)

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1464137

Test results:
```
λ pytest -v tests/foreman/api/test_permission.py -k 'UserRoleTestCase'
===================================================================================== test session starts =====================================================================================
platform linux2 -- Python 2.7.13, pytest-3.1.0, py-1.4.33, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
metadata: {'Python': '2.7.13', 'Platform': 'Linux-4.11.3-1-ARCH-x86_64-with-glibc2.2.5', 'Packages': {'py': '1.4.33', 'pytest': '3.1.0', 'pluggy': '0.4.0'}, 'Plugins': {'cov': '2.5.1', 'xdist': '1.16.0', 'html': '1.15.1', 'services': '1.2.1', 'mock': '1.6.0', 'metadata': '1.5.0'}}
rootdir: /home/qui/code/robottelo, inifile:
plugins: xdist-1.16.0, services-1.2.1, mock-1.6.0, metadata-1.5.0, html-1.15.1, cov-2.5.1
collected 7 items 
2017-06-22 17:02:21 - conftest - DEBUG - Collected 7 test cases
tests/foreman/api/test_permission.py::UserRoleTestCase::test_positive_check_create PASSED
tests/foreman/api/test_permission.py::UserRoleTestCase::test_positive_check_delete PASSED
tests/foreman/api/test_permission.py::UserRoleTestCase::test_positive_check_read PASSED
tests/foreman/api/test_permission.py::UserRoleTestCase::test_positive_check_update PASSED

============================================ 3 tests deselected ============================================
======================== 4 passed, 3 deselected, 1 warnings in 159.24 seconds =========================
```

### CODE REVIEW

File: tests/foreman/api/test_permission.py [hunk 1]
```diff
@@ -249,15 +261,31 @@ def test_positive_check_create(self):
...
                     entity_cls(self.cfg).create()
                 self.give_user_permission(
                     _permission_name(entity_cls, 'create')
                 )
-                entity_id = entity_cls(self.cfg).create_json()['id']
-                entity_cls(id=entity_id).read()  # As admin user.
+                entity = entity_cls(self.cfg)
+                entity_fields = entity.get_fields()
+                if 'organization' in entity_fields:
+                    if entity_fields['organization'] is OneToManyField:
```
Comment (svtkachenko) [inline tests/foreman/api/test_permission.py]: Not sure about this one, but I wasn't able to find a way to distinguish between entities that expect list of organizations and the ones that expect only one organization.

Comment (ldjebran) [inline tests/foreman/api/test_permission.py]: think it must be:
```python
 isinstance(entity_fields['organization'], OneToManyField)
```

File: tests/foreman/api/test_permission.py [hunk 2]
```diff
@@ -248,16 +260,35 @@ def test_positive_check_create(self):
...
+                            entity_fields['organization'], OneToManyField):
+                        entity.organization = [self.org]
+                    else:
+                        entity.organization = self.org
+                if 'location' in entity_fields:
+                    # Currently entities with both org and loc are affected by
+                    # bug. Skip to the next entity
+                    if bz_bug_is_open('1464137'):
+                        continue
+                    entity.location = [self.loc]
```
Comment (abalakh) [inline tests/foreman/api/test_permission.py]: Technically it's not an entity with both fields but entity which has location, according to your `'location' in entity_fields` check. Also you're not covering the case of OneToOneField for location

File: tests/foreman/api/test_permission.py [hunk 3]
```diff
@@ -272,13 +303,21 @@ def test_positive_check_read(self):
...
+        for entity_cls in (
+                entities.Architecture,
+                entities.Domain,
+                entities.ActivationKey):
             with self.subTest(entity_cls):
-                entity_id = entity_cls().create().id
+                entity = entity_cls()
+                if ('organization' in entity.get_fields()
+                        and 'location' in entity.get_fields()):
+                    entity.organization = [self.org]
```
Comment (abalakh) [inline tests/foreman/api/test_permission.py]: So for this and all the later tests `OneToManyField` checks aren't needed anymore? I don't get it, the list of entities is the same, why different approach?

**Pull request merged**